fastly/resource_fastly_service_v1_snippet_test.go

```diff
@@ -25,6 +25,13 @@ func TestResourceFastlyFlattenSnippets(t *testing.T) {
 					Priority: 110,
 					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
 				},
+				{
+					Name:     "recv_test",
+					Type:     gofastly.SnippetTypeRecv,
+					Priority: 110,
+					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
+					Dynamic:  1,
+				},
```

Comment:
The second object of the remote field will never be compared against a local object! In order to test that the `flattenSnippets ` function ignores `dynamic` snippets the test should look like:

```
func TestResourceFastlyFlattenSnippets(t *testing.T) {

	cases := []struct {
		remote []*gofastly.Snippet
		local  []map[string]interface{}
	}{
		{
			remote: []*gofastly.Snippet{
				{
					Name:     "recv_test",
					Type:     gofastly.SnippetTypeRecv,
					Priority: 110,
					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
				},
			},
			local: []map[string]interface{}{
				{
					"name":     "recv_test",
					"type":     gofastly.SnippetTypeRecv,
					"priority": 110,
					"content":  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
				},
			},
		},
		{
			remote: []*gofastly.Snippet{
				{
					Name:     "recv_test",
					Type:     gofastly.SnippetTypeRecv,
					Priority: 110,
					Content:  "if ( req.url ) {\n set req.http.my-snippet-test-header = \"true\";\n}",
					Dynamic:  1,
				},
			},
			local: []map[string]interface{}(nil),
		},
	}

	for _, c := range cases {
		out := flattenSnippets(c.remote)
		if !reflect.DeepEqual(out, c.local) {
			t.Fatalf("Error matching:\nexpected: %#v\n got: %#v", c.local, out)
		}
	}

}
```

Reply:
Updated unit test